extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetInstructions.java

```diff
@@ -520,9 +669,17 @@ public static class Builder {
         private Object specialInstructions;
         private boolean generateMetadataFiles = DEFAULT_GENERATE_METADATA_FILES;
         private String baseNameForPartitionedParquetData = DEFAULT_BASE_NAME_FOR_PARTITIONED_PARQUET_DATA;
+        private Optional<ParquetFileLayout> fileLayout = Optional.empty();
+        private Optional<TableDefinition> tableDefinition = Optional.empty();
+        private Optional<Collection<String[]>> indexColumns = Optional.empty();
```

Comment:
It's very odd to see `Optional` fields; the only time when they have value is when you need to differentiate between "emtpy" / not-set and `null`. I think in this case, you can just have non-optional values that have `null`? And when you need to pass them off, use `Optional.ofNullable`?

Reply:
We need to distinguish. "not set" means "infer from tables".

Reply:
Why do we need optional for this? If you want to explicitly set as "don't use index", just use an empty list.

Reply:
In the current code,
- Index columns not set means if the table has any index columns, write them down, else don't.
- Empty collection means don't write any index columns, irrespective of whether the table has any index columns
- Non empty value means only write down the provided index columns,  irrespective of whether the table has any index columns

Reply:
So you are proposing:
- null means if the table has any index columns, write them down, else don't.
- Empty collection means don't write any index columns, irrespective of whether the table has any index columns
- Non empty collection means only write down the provided index columns, irrespective of whether the table has any index columns

Reply:
To be clear, it's very common to have Optional as getters, just not as fields. The typical pattern is

```java
private final X field;

public Optional<X> getField() { return Optional.ofNullable(field); }
```